xsnippet_api/resources/snippets.py

```diff
@@ -16,44 +16,101 @@
 from ..application import endpoint
 
 
-_schema = {
-    'id': {'type': 'integer', 'readonly': True},
-    'title': {'type': 'string'},
-    'content': {'type': 'string', 'required': True},
-    'syntax': {'type': 'string'},
-    'tags': {'type': 'list', 'schema': {'type': 'string', 'regex': '[\w_-]+'}},
-    'is_public': {'type': 'boolean'},
-    'author_id': {'type': 'integer', 'readonly': True},
-    'created_at': {'type': 'datetime', 'readonly': True},
-    'updated_at': {'type': 'datetime', 'readonly': True},
-}
+class InvalidId(Exception):
```

Comment:
It's sort of local exception: it's used only there and outer world knows nothing about that. But maybe I overdesigned here and we should put it into exceptions.py.